src/librustc/hir/lowering.rs

```diff
@@ -3256,7 +3256,7 @@ impl<'a> LoweringContext<'a> {
             // If `<pattern>` is a simple ident, then it is lowered to a single
             // `let <pattern> = <pattern>;` statement as an optimization.
             for (index, argument) in decl.inputs.iter().enumerate() {
-                let argument = this.lower_arg(argument);
+                let argument = this.lower_param(argument);
```

Comment:
Variables should probably be converted as well

Reply:
Changed